tdms/include/field.h

```diff
@@ -161,3 +174,30 @@ class MagneticField: public Field{
 private:
     std::complex<double> phasor_norm(double f, int n, double omega, double dt, int Nt) override;
 };
+
+/**
+ * Structure to hold a field and allow saving it to a file
+ */
+class TDFieldExporter2D{
+
+public:
+  mxArray* matlab_array = nullptr;
+  double** array = nullptr;
+  char* folder_name = nullptr;
+
+  /**
+   * Allocate the arrays to hol the field
+   */
+  void allocate(int I, int J);
```

Comment:
Use `I_tot` and `J_tot` here to be consistent with variable names that denote size of grids/arrays?

Reply:
This actually allocates IxJ not I+1xJ+1 so may be better to leave(?)

Reply:
Maybe `nI` or `nJ` instead then? Something to indicate these variables are array sizes and not array indices.